.github/workflows/docker.yml

```diff
@@ -0,0 +1,27 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build the CWLViewer Java Maven docker image
+        run: docker build -t cwlviewer-gh-actions .
+      - name: Confirm we can execute java in the container
+        run: docker run -ti cwlviewer-gh-actions java -version
+      - name: Validate docker-compose
+        run: docker-compose -f docker-compose.yml config
+      - name: Create the docker-compose services
+        run: docker-compose -f docker-compose.yml up --no-start
+      - name: Start the docker-compose services
+        run: docker-compose -f docker-compose.yml start
+      - name: Stop the docker-compose services
+        run: docker-compose -f docker-compose.yml stop
```

Comment:
This should cover testing the `Dockerfile` and `docker-compose.yml`. Initially I was going to use the official Docker GitHub Actions, but couldn't find a way to just build it without pushing. And we can add it later any time if we find value in doing so, https://github.com/docker/build-push-action/ (found a few issues in their docs, sending a PR in a few minutes).

Reply:
Maybe, but that could complicate the test setup, I think.

There were some tests that started an instance of MongoDB. We migrated to PostgreSQL, and these tests were upgraded to now start a PostgreSQL instance with Docker for each test, re-creating the schema - https://github.com/common-workflow-language/cwlviewer/blob/8bf8a371378d90913dbf1a3f4a0981e3a8dea3e4/src/test/java/org/commonwl/view/workflow/QueuedWorkflowRepositoryTest.java#L39-L43

I think it is still doable as it is not impossible to run [Docker inside Docker](https://devopscube.com/run-docker-in-docker/). I never did that, but I suspect it could end up complicating to troubleshoot our tests, plus the effort to configure it.

Another option would be to run a subset of the tests (the ones that do not use Docker) in the cluster. The Jena image wouldn't be used, nor the PostgreSQL…

Reply:
My idea was to just build the image from `Dockerfile` to avoid the incorrect `java` executable location. But I realized we could also test Docker Compose in GH Actions. But if that's simpler, @mr-c, maybe we can remove the Docker Compose, leave the tests for `Dockerfile` only, and open another issue about testing the application within the Docker Compose cluster. That does sound like a good idea to perform some tests in the complete system, maybe even using something like Nightwatch/Cypress/Selenium/etc.